### PR TITLE
RESOURCE-233 AWS::ApiGateway::DomainName

### PR DESCRIPTION
**Pull request by soumyo13**
Signed-off-by: Soumyodeep Karmakar <soumyo.k13@gmail.com>

### Description

1. It contains all the singular and plural resources.
2. Added the resource_id in the singular resource.
3. Modified the docs according to the new format. 

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill a box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [ ] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Review by sa-progress — APPROVED** [review]
LGTM

File: docs-chef-io/content/inspec/resources/aws_api_gateway_domain_name.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+: **Field**: `mutual_tls_authentication.truststore_version`
+
+`mutual_tls_authentication_truststore_warnings`
+: A list of warnings that API Gateway returns while processing your truststore. Invalid certificates produce warnings. Mutual TLS is still enabled, but some clients might not be able to access your API. To resolve warnings, upload a new truststore to S3, and then update you domain name to use the new version.
+
+: **Field**: `mutual_tls_authentication.truststore_warnings`
+
+`ownership_verification_certificate_arn`
+: The ARN of the public certificate issued by ACM to validate ownership of your custom domain. Only required when configuring mutual TLS and using an ACM imported or private CA certificate ARN as the regionalCertificateArn.
```
Comment (dkumaras) [inline docs-chef-io/content/inspec/resources/aws_api_gateway_domain_name.md:135]: @soumyo13 - do we need to include the **RegionalCertificateArn** field?

Comment (dkumaras) [reply docs-chef-io/content/inspec/resources/aws_api_gateway_domain_name.md:135]: and, is this a one word here: https://docs.aws.amazon.com/cloudfront/index.html

same in plural resource too!

File: docs-chef-io/content/inspec/resources/aws_api_gateway_domain_name.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+: **Field**: `endpoint_configuration.types`
+
+`endpoint_configuration_vpc_endpoint_ids`
+: A list of VpcEndpointIds of an API (RestApi) against which to create Route53 ALIASes. It is only supported for PRIVATE endpoint type.
+
+: **Field**: `endpoint_configuration.vpc_endpoint_ids`
+
+`domain_name_status`
+: The status of the DomainName migration. The valid values are AVAILABLE and UPDATING . If the status is UPDATING , the domain cannot be modified further until the existing operation is complete. If it is AVAILABLE , the domain can be updated.
```
Comment (dkumaras) [inline docs-chef-io/content/inspec/resources/aws_api_gateway_domain_name.md:100]: @soumyo13 - DomainName - one word?

**Review by dkumaras — COMMENTED** [review]
Pls check & confirm.

File: docs-chef-io/content/inspec/resources/aws_api_gateway_domain_name.md (hunk 1 above)
Comment (soumyo13) [reply docs-chef-io/content/inspec/resources/aws_api_gateway_domain_name.md]: We can remove the second line from both the docs files.

File: docs-chef-io/content/inspec/resources/aws_api_gateway_domain_name.md (hunk 2 above)
Comment (soumyo13) [reply docs-chef-io/content/inspec/resources/aws_api_gateway_domain_name.md]: we can make it to two different words.

**Comment by soumyo13** [comment]
> Pls check & confirm.

Looks good to me! Please approve it.

**Pull request merged**